### PR TITLE
Precise preg_match_all return type

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
> afaik positive-int is being deprecated, should be `int<0, max>` where you use that in a union with 0

I updated the notation then

**Comment by VincentLanglet** [comment]
Friendly ping @weirdan @orklah if you have time to review :)

File: stubs/CoreGenericFunctions.phpstub [hunk 1]
```diff
@@ -1249,7 +1249,7 @@ function preg_replace_callback($pattern, $callback, $subject, int $limit = -1, &
  *              )
  *          )
  *        ) $matches
- * @return int|false
+ * @return 0|positive-int|false
```
Comment (weirdan) [inline stubs/CoreGenericFunctions.phpstub:1252]: `@return int<0,max>|false` perhaps?

Comment (VincentLanglet) [reply stubs/CoreGenericFunctions.phpstub]: Indeed, changed

**Comment by weirdan** [comment]
Thanks!

**Pull request merged**